dataprofiler/tests/data_readers/test_data_utils.py

```diff
@@ -36,3 +37,125 @@ def test_file_UTF_encoding_detection(self):
             detected_encoding = \
                 data_utils.detect_file_encoding(file_path=input_file["path"])
             self.assertEqual(detected_encoding.lower(), input_file["encoding"])
+
+    def test_nth_loc_detection(self):
+        """
+        Tests the ability for the `data_utils.find_nth_location` to detect the
+        nth index of a search_query in a string. 
+        """
+        # Input args: string, query, n
+        # Expected results: index, occurances
+        test_queries = [
+            dict(string="This is a test.", query=".", n=1, index=14, occurances=1),
+            dict(string="This is a test\n", query="\n", n=1, index=14, occurances=1),
+            dict(string="This is a test\nThis is a second test\n",
+                 query="\n", n=0, index=-1, occurances=0),
+            dict(string="This is a test\nThis is a second test\n",
+                 query="\n", n=2, index=36, occurances=2),
+            dict(string='t', query="t", n=1, index=0, occurances=1),
+            dict(string='s', query="t", n=1, index=1, occurances=0),
+            dict(string="This is a test\nThis is a second test\n\n",
+                 query="\n", n=3, index=37, occurances=3),
+            dict(string="This is a test\nThis is a second test\n\nTest\n",
+                 query="\n", n=5, index=43, occurances=4),
+            dict(string="", query="\n", n=3, index=-1, occurances=0),
+        ]
+
+        for q in test_queries:
+            self.assertEqual(
+                data_utils.find_nth_loc(q['string'], q['query'], q['n']),
+                (q['index'], q['occurances'])
+            )
+
+    def test_load_as_str_from_file(self):
+        """
+        Tests if the load_as_str_file function can appropriately load files 
+        thresholded by bytes or max lines.
+        """
+
+        test_dir = os.path.join(test_root_path, 'data')
+        
+        iris_32bit_filepath = os.path.join(test_dir, 'csv/iris-utf-32.csv')
+        iris_32bit_first_5  = ""
+        with open(iris_32bit_filepath, encoding='utf-32') as f:
+            iris_32bit_first_5 = ''.join(list(islice(f, 5)))[:-1] # remove final \n
+
+        input_files = [
+            dict(
+                path=os.path.join(test_dir, 'csv/empty.csv'),
+                encoding='utf-8', max_lines=5, max_bytes=65536,
+                chunk_size_bytes=1024,
+                results = ""
+            ),dict(
+                path=os.path.join(test_dir, 'csv/iris-utf-8.csv'),
+                encoding='utf-8', max_lines=5, max_bytes=65536,
+                chunk_size_bytes=1024,
+                results = """Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
+1,5.1,3.5,1.4,0.2,Iris-setosa
+2,4.9,3.0,1.4,0.2,Iris-setosa
+3,4.7,3.2,1.3,0.2,Iris-setosa
+4,4.6,3.1,1.5,0.2,Iris-setosa"""
```

Comment:
i find this format extremely confusing. did we have to use `"""` to do this?
something like:
```python
"Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species\n"
"1,5.1,3.5,1.4,0.2,Iris-setosa\n"
"2,4.9,3.0,1.4,0.2,Iris-setosa\n"
"3,4.7,3.2,1.3,0.2,Iris-setosa\n"
"4,4.6,3.1,1.5,0.2,Iris-setosa\n"
```
might be more readable